### ISSUE TITLE
[Question] Obvious "gap / void" along the top edge when content moves upward during interpolation — is this expected?

### ISSUE THREAD

**Issue by whyb** [KAIST-VICLab/BiM-VFI#6]
# [Question] Obvious "gap / void" along the top edge when content moves upward during interpolation — is this expected?

First of all, thank you very much for sharing **BiM-VFI (CVPR 2025)** and for releasing the official code and pre-trained weights. The idea of Bidirectional Motion Field-guided interpolation for videos with *non-uniform motions* is very interesting and inspiring, and the demo pipeline (`main.py --cfg cfgs/bim_vfi_demo.yaml`) is easy to use. We really appreciate your great work. 🙏

---

## 1. Environment / Setup

- PyTorch **2.4.1** (we ran the official demo on CPU; `cupy` is unavailable here, so the cost-volume **forward** was replaced by an element-wise equivalent pure-PyTorch implementation — forward numerics are identical, only the CUDA/cupy kernel is not used).
- Input: two **4K (2160 × 3840)** frames extracted from a 60 fps test video, interpolated at **t = 0.5**.
- We followed the README **Demo** section: put the frames as a video folder and ran

```bash
python main.py --cfg cfgs/bim_vfi_demo.yaml
```

The official demo picks `pyr_level` from the frame height (`>= 2160 -> pyr_level = 7` in `modules/models/inference_video.py`), and the model forward keeps `antialias=True` downsampling.

## 2. Input frames

**Fig. 1 — Input frame A** (`input3/1.jpg`, 2160 × 3840)

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/122025d3-f6b2-40e0-b3c5-62654f3900ca" />

**Fig. 2 — Input frame B** (`input3/3.jpg`, 2160 × 3840)

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/52543cce-7886-41dd-a3db-6d79bc48fdf7" />

## 3. Interpolated result (t = 0.5)

**Fig. 3 — BiM-VFI output at t = 0.5** (`input3/2_pytorch.jpg`, produced by the official demo above)

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/5b1c1573-8421-4a48-b255-f82acdb58447" />

**Fig. 4 — RIFE output at t = 0.5** (`input3/2_rife.jpg`, produced with the RIFE model on the same pair, for comparison)

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/a090f037-10b6-414a-a909-7ea347321c7d" />

**Fig. 5 — Side-by-side comparison** (Input A | Input B | BiM-VFI | RIFE)

<img width="3840" height="540" alt="Image" src="https://github.com/user-attachments/assets/8a8687e8-6d88-49f2-ada2-3c3c68b99451" />

**Fig. 6 — Zoom on the region where the two models differ most** (around the moving object boundary)

<img width="1800" height="300" alt="Image" src="https://github.com/user-attachments/assets/2941e756-5ae7-41a7-8237-ede792d22b63" />

---

## 4. Observed problem / our doubt

In the official demo interpolation of our test video, **during the segments where the content/scene moves upward, an obvious "gap / void" appears along the TOP edge of the interpolated frames** — it looks like a blank or mismatched band right at the image border (see Fig. 7; Figs. 1–6 show the same pipeline on a still pair for reference).

Compared with **RIFE** on the same content, the border area looks less clean in the BiM-VFI output: RIFE keeps the top boundary more consistent, while BiM-VFI tends to show this empty seam when motion points out of the frame.

Some numbers on the attached still pair (t = 0.5, 2160 × 3840, full image, no border crop):

| comparison | PSNR |
|---|---|
| BiM-VFI output vs RIFE output | ≈ 38.3 dB |
| BiM-VFI output vs input A / input B | ≈ 24.8 / 24.8 dB |
| RIFE output vs input A / input B | ≈ 24.8 / 24.8 dB |

The two models are globally close, but locally (at moving-object boundaries / frame borders) the outputs differ clearly (Fig. 6).

### Our questions

1. Is this **top-edge "gap" under out-of-frame (disocclusion) motion an expected limitation** of the current framework (e.g., caused by constant zero-padding in `InputPadder`, `grid_sample(padding_mode='zeros')`, and/or normalization over the padded canvas)? Is there a recommended way to reduce it — e.g., mirror/reflect padding before feeding the network, or cropping the border region of the output?
2. In the paper/benchmarks, are border areas usually **center-cropped before computing metrics**? We would like to know whether this boundary behavior is considered normal and simply excluded in evaluation.
3. We also exported an **ONNX** model (`bimvfi.onnx`, FP32, dynamic H/W) and wrote a small ONNX Runtime inference script (`infer_onnx.py`, both attached below). The ONNX result is very close to the PyTorch result on still pairs (PSNR ≈ 45 dB, mean abs diff ≈ 0.37/255), but we would like to confirm: **is a small PyTorch-vs-ONNX difference of this magnitude expected**, and does the same top-edge behavior apply to the ONNX deployment as well?

---

## 5. Reproduction steps (for the still-pair example)

1. Put the two frames as a video folder, e.g.:

```
assets/demo_input3/input3/00000.jpg   # = input frame A
assets/demo_input3/input3/00001.jpg   # = input frame B
```

2. Demo config (`cfgs/demo_input3.yaml`), same as the official demo config but with:

```yaml
exp_name: original
mode: demo
resume: pretrained/bim_vfi.pth
model:
  name: bim_vfi
  args:
    pyr_level: 3
    feat_channels: 32
demo_dataset:
  name: demo
  args:
    root_path: ./assets/demo_input3
```

3. Run:

```bash
python main.py --cfg cfgs/demo_input3.yaml
```

4. The official demo performs 8× interpolation (t = 1/8 … 7/8). The **t = 0.5** frame is `0000004.jpg` under `save/<exp_name>/output/demo/input3/`.

---

## 6. Attachments

[BiM-VFI_test_pictures.zip](https://github.com/user-attachments/files/31818842/BiM-VFI_test_pictures.zip)

[BiM-VFI.7z.rename to 7z.zip](https://github.com/user-attachments/files/31818811/BiM-VFI.7z.rename.to.7z.zip)

---

Thank you again for the excellent work, and thank you in advance for any clarification! We are happy to provide the full test video clip or any other material if it helps.

Best regards,
whyb
